### DISCUSSION TITLE
Please build fira code for alternative character variants.

### DISCUSSION THREAD

**Discussion by MadadiSaeed** [tonsky/FiraCode#1295] [General]
Hi. I cloned the repository and I tried to build fira code for alternative character variants but I got so many errors in running `./script/bootstrap_linux.sh`. I like alternative variants more than default variants. I am trying to do this mainly because I want to use it in my vscode terminal. vscode terminal does not recognize alternative variants when enabling it. Thank you so much. I am using ubuntu 20.04 lts.

**Comment by tonsky** [comment]
Which variants do you need?

**Reply by MadadiSaeed** [reply]
ss01, ss03, ss04, ss07, zero. 
I like these variants mainly because: 
1. I love how these variants look.
2. I wanna use fira code with these variants in terminal as default font.

Thank you so much.

**Reply by alexeyten** [reply]
You could try configure fontconfig to use these stylesets see https://github.com/tonsky/FiraCode/issues/905#issuecomment-561129223 for example

**Reply by MadadiSaeed** [reply]
first: since I am using elementary (ubuntu based) os and the default terminal app name is `io.elementary.terminal`, I tried both `gnome-terminal` and `io.elementary.terminal` and rebooted my system, but none of these worked. second: I do not care about ligatures since most off the apps that I wanna use do not support ligatures or this type of font configuration. I just need those characters as default character in fira code. I hope in the future fira code community help main developer of this lovely font to create a website to select their ligatures and character variants as they wish and build the font from that, like monoid font that has a simple website to do this task.

**Reply by tonsky** [reply]
Unfortunately, I can’t build it right now, might take some time — something is broken and needs troubleshooting. Will look into it later

**Reply by alexeyten** [reply]
This config works for me in terminal and normal apps.
```xml
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<fontconfig>
    <match target="scan">
        <test name="family" compare="eq">
            <string>Fira Code</string>
        </test>
        <edit name="spacing" mode="assign_replace">
            <const>mono</const>
        </edit>
    </match>

    <match target="font">
        <test name="family" compare="eq">
            <string>Fira Code</string>
        </test>
        <edit name="fontfeatures" mode="append">
            <!-- 0 r & $ =~ !~ -->
            <string>zero on</string>
            <string>ss01 on</string>
            <string>ss03 on</string>
            <string>ss04 on</string>
            <string>ss07 on</string>
        </edit>
    </match>
</fontconfig>
```

![image](https://user-images.githubusercontent.com/44377/135284837-c51ee5f1-657c-440c-8c1e-2c3a475b7c6c.png)

**Reply by MadadiSaeed** [reply]
I used this font config and It worked in both elementary os editor and terminal application But do you have any idea Why this is not working in vscode terminal?

**Reply by alexeyten** [reply]
You could try to use https://twardoch.github.io/fonttools-opentype-feature-freezer/ to “freeze” ssNN features

**Reply by MadadiSaeed** [reply]
Thank you so much. It is working now. You saved my day.

**Comment by MadadiSaeed** [comment]
@tonsky Thank you so much. I appreciate your help.
